homework/App.js

```diff
@@ -12,18 +12,31 @@ class App {
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
-    const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
-
     try {
+      const root = document.getElementById('root');
+      const header = Util.createAndAppend('header', root, { class: 'header' });
+      Util.createAndAppend('div', root, {
+        class: 'container',
+        id: 'container',
+      });
+      const headerDiv = Util.createAndAppend('div', header, { class: 'headerDiv' });
+      Util.createAndAppend('span', headerDiv, { text: 'HYF Repositories', id: 'headerSpan' });
+      const select = Util.createAndAppend('select', headerDiv, { class: 'select' });
+
       const repos = await Util.fetchJSON(url);
+      repos.sort((a, b) => a.name.localeCompare(b.name));
       this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.repos.forEach(repo => Util.createAndAppend('option', select, { text: repo.name() }));
+      let index = select.selectedIndex;
+      this.fetchContributorsAndRender(index);
```

Comment:
I would just forget about creating an extra variable `index` here, as you have no other use for this variable other than to pass it as an argument to `fetchContributorsAndRender()`:

```js
this.fetchContributorsAndRender( select.selectedIndex);
```